src/test/scala/model/EnemyModelTest.scala

```diff
@@ -1,38 +1,90 @@
-  }
-  "A enemy" should "filter the interesting entities" in {
-    val basicZombie: Enemy = Zombie((1, LanesLength / 2))
-    val firstTurretInFirstLane: Plant = PeaShooter((1, LanesLength /2 - 1))
-    val secondTurretInSecondLane: Plant = PeaShooter((2, LanesLength * 0.75))
-    val firstZombieFirstLane: Enemy = Zombie((1, LanesLength))
-    val thirdZombieInSecondLane: Enemy = Zombie((2, LanesLength))
-    val entities: List[Entity] = List(firstTurretInFirstLane, secondTurretInSecondLane,
-      firstZombieFirstLane, thirdZombieInSecondLane)
-    assert(entities.filter(basicZombie.isInterestedIn) == List(firstTurretInFirstLane))
+  private val testingLane = 1
+  private val otherLane = 2
+  private val zombie: Troop = Troops.ofType[Zombie] withPosition (testingLane, 80)
+  private val fastZombie: Troop = Troops.ofType[FastZombie] withPosition (testingLane, 80)
+  private val warriorZombie: Troop = Troops.ofType[WarriorZombie] withPosition (testingLane, 80)
+
+  private val plantInOtherLane: Troop = Troops.ofType[PeaShooter] withPosition (otherLane, 10)
+  private val plantInTheSameLane: Troop = Troops.ofType[PeaShooter] withPosition (testingLane, 10)
+  private val plantInRange: Troop = Troops.ofType[PeaShooter] withPosition (testingLane, 75)
+
+  private val dummyPlant: Troop = Troops.ofType[PeaShooter] withPosition (otherLane, 50)
+  private val dummyZombie: Troop = Troops.ofType[Zombie] withPosition (otherLane, 50)
+  private val dummyBullet: Bullet = PeaBullet(0,0)
+
+
+  "A Zombie" should "be in Moving state if it can't attack any plant" in {
+    zombie.update(FiniteDuration(16, "milliseconds"), List()).state shouldBe Moving
+  }
+  "A Zombie" should "enter attacking state if it can attack an enemy" in {
+    zombie.update(FiniteDuration(16, "milliseconds"), List(plantInTheSameLane)).state shouldBe Attacking
+  }
+  "A Zombie" should "filter the interesting entities" in {
+    List(dummyPlant, plantInTheSameLane, dummyZombie, dummyBullet, plantInRange) filter zombie.isInterestedIn shouldBe List(plantInRange)
+  }
+  "A Zombie" should "lose HPs after getting hit" in {
+    (zombie collideWith dummyBullet).life should be < zombie.life
+  }
+  "A Zombie" should "die if reaches 0 or less HP" in {
+    (zombie withLife 25 collideWith dummyBullet).state shouldBe Dead
+  }
+  "A Zombie" should "not have interests" in {
+    List(dummyZombie, plantInTheSameLane, dummyBullet) filter zombie.isInterestedIn shouldBe List.empty
+  }
+
+  "A FastZombie" should "be in Moving state if it can't attack any plant" in {
+    fastZombie.update(FiniteDuration(16, "milliseconds"), List()).state shouldBe Moving
+  }
+  "A FastZombie" should "enter attacking state if it can attack an enemy" in {
+    fastZombie.update(FiniteDuration(16, "milliseconds"), List(plantInTheSameLane)).state shouldBe Attacking
   }
+  "A FastZombie" should "filter the interesting entities" in {
```

Comment:
This feels like a copy paste from the Zombie tests above
Why are you testing again the isInterestIn method? It's the same for every zombie..